### DISCUSSION TITLE
Implement the 100+ rules from eslint-plugin-unicorn

### DISCUSSION THREAD

**Discussion by flexiworld** [rome/tools#4756] [Suggestions]
Implement the 100+ rules from eslint-plugin-unicorn: https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main

✅ Set in the `recommended` [configuration](https://github.com/sindresorhus/eslint-plugin-unicorn#preset-configs).\
🔧 Automatically fixable by the [`--fix` CLI option](https://eslint.org/docs/user-guide/command-line-interface#--fix).\
💡 Manually fixable by [editor suggestions](https://eslint.org/docs/developer-guide/working-with-rules#providing-suggestions).

| Name                                                                                             | Description                                                                                                                                                                                                       | 💼 | 🔧 | 💡 |
| :----------------------------------------------------------------------------------------------- | :---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | :- | :- | :- |
| [better-regex](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/better-regex.md)                                                       | Improve regexes by making them shorter, consistent, and safer.                                                                                                                                                    | ✅  | 🔧 |    |
| [catch-error-name](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/catch-error-name.md)                                               | Enforce a specific parameter name in catch clauses.                                                                                                                                                               | ✅  | 🔧 |    |
| [consistent-destructuring](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/consistent-destructuring.md)                               | Use destructured variables over properties.                                                                                                                                                                       | ✅  | 🔧 | 💡 |
| [consistent-function-scoping](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/consistent-function-scoping.md)                         | Move function definitions to the highest possible scope.                                                                                                                                                          | ✅  |    |    |
| [custom-error-definition](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/custom-error-definition.md)                                 | Enforce correct `Error` subclassing.                                                                                                                                                                              |    | 🔧 |    |
| [empty-brace-spaces](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/empty-brace-spaces.md)                                           | Enforce no spaces between braces.                                                                                                                                                                                 | ✅  | 🔧 |    |
| [error-message](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/error-message.md)                                                     | Enforce passing a `message` value when creating a built-in error.                                                                                                                                                 | ✅  |    |    |
| [escape-case](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/escape-case.md)                                                         | Require escape sequences to use uppercase values.                                                                                                                                                                 | ✅  | 🔧 |    |
| [expiring-todo-comments](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/expiring-todo-comments.md)                                   | Add expiration conditions to TODO comments.                                                                                                                                                                       | ✅  |    |    |
| [explicit-length-check](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/explicit-length-check.md)                                     | Enforce explicitly comparing the `length` or `size` property of a value.                                                                                                                                          | ✅  | 🔧 | 💡 |
| [filename-case](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/filename-case.md)                                                     | Enforce a case style for filenames.                                                                                                                                                                               | ✅  |    |    |
| [import-style](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/import-style.md)                                                       | Enforce specific import styles per module.                                                                                                                                                                        | ✅  |    |    |
| [new-for-builtins](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/new-for-builtins.md)                                               | Enforce the use of `new` for all builtins, except `String`, `Number`, `Boolean`, `Symbol` and `BigInt`.                                                                                                           | ✅  | 🔧 |    |
| [no-abusive-eslint-disable](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/no-abusive-eslint-disable.md)                             | Enforce specifying rules to disable in `eslint-disable` comments.                                                                                                                                                 | ✅  |    |    |
| [no-array-callback-reference](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/no-array-callback-reference.md)                         | Prevent passing a function reference directly to iterator methods.                                                                                                                                                | ✅  |    | 💡 |
| [no-array-for-each](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/no-array-for-each.md)                                             | Prefer `for…of` over the `forEach` method.                                                                                                                                                                        | ✅  | 🔧 | 💡 |
| [no-array-method-this-argument](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/no-array-method-this-argument.md)                     | Disallow using the `this` argument in array methods.                                                                                                                                                              | ✅  | 🔧 | 💡 |
| [no-array-push-push](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/no-array-push-push.md)                                           | Enforce combining multiple `Array#push()` into one call.                                                                                                                                                          | ✅  | 🔧 | 💡 |
| [no-array-reduce](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/no-array-reduce.md)                                                 | Disallow `Array#reduce()` and `Array#reduceRight()`.                                                                                                                                                              | ✅  |    |    |
| [no-await-expression-member](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/no-await-expression-member.md)                           | Disallow member access from await expression.                                                                                                                                                                     | ✅  | 🔧 |    |
| [no-console-spaces](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/no-console-spaces.md)                                             | Do not use leading/trailing space between `console.log` parameters.                                                                                                                                               | ✅  | 🔧 |    |
| [no-document-cookie](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/no-document-cookie.md)                                           | Do not use `document.cookie` directly.                                                                                                                                                                            | ✅  |    |    |
| [no-empty-file](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/no-empty-file.md)                                                     | Disallow empty files.                                                                                                                                                                                             | ✅  |    |    |
| [no-for-loop](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/no-for-loop.md)                                                         | Do not use a `for` loop that can be replaced with a `for-of` loop.                                                                                                                                                | ✅  | 🔧 |    |
| [no-hex-escape](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/no-hex-escape.md)                                                     | Enforce the use of Unicode escapes instead of hexadecimal escapes.                                                                                                                                                | ✅  | 🔧 |    |
| [no-instanceof-array](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/no-instanceof-array.md)                                         | Require `Array.isArray()` instead of `instanceof Array`.                                                                                                                                                          | ✅  | 🔧 |    |
| [no-invalid-remove-event-listener](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/no-invalid-remove-event-listener.md)               | Prevent calling `EventTarget#removeEventListener()` with the result of an expression.                                                                                                                             | ✅  |    |    |
| [no-keyword-prefix](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/no-keyword-prefix.md)                                             | Disallow identifiers starting with `new` or `class`.                                                                                                                                                              |    |    |    |
| [no-lonely-if](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/no-lonely-if.md)                                                       | Disallow `if` statements as the only statement in `if` blocks without `else`.                                                                                                                                     | ✅  | 🔧 |    |
| [no-negated-condition](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/no-negated-condition.md)                                       | Disallow negated conditions.                                                                                                                                                                                      | ✅  | 🔧 |    |
| [no-nested-ternary](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/no-nested-ternary.md)                                             | Disallow nested ternary expressions.                                                                                                                                                                              | ✅  | 🔧 |    |
| [no-new-array](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/no-new-array.md)                                                       | Disallow `new Array()`.                                                                                                                                                                                           | ✅  | 🔧 | 💡 |
| [no-new-buffer](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/no-new-buffer.md)                                                     | Enforce the use of `Buffer.from()` and `Buffer.alloc()` instead of the deprecated `new Buffer()`.                                                                                                                 | ✅  | 🔧 | 💡 |
| [no-null](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/no-null.md)                                                                 | Disallow the use of the `null` literal.                                                                                                                                                                           | ✅  | 🔧 | 💡 |
| [no-object-as-default-parameter](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/no-object-as-default-parameter.md)                   | Disallow the use of objects as default parameters.                                                                                                                                                                | ✅  |    |    |
| [no-process-exit](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/no-process-exit.md)                                                 | Disallow `process.exit()`.                                                                                                                                                                                        | ✅  |    |    |
| [no-static-only-class](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/no-static-only-class.md)                                       | Disallow classes that only have static members.                                                                                                                                                                   | ✅  | 🔧 |    |
| [no-thenable](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/no-thenable.md)                                                         | Disallow `then` property.                                                                                                                                                                                         | ✅  |    |    |
| [no-this-assignment](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/no-this-assignment.md)                                           | Disallow assigning `this` to a variable.                                                                                                                                                                          | ✅  |    |    |
| [no-typeof-undefined](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/no-typeof-undefined.md)                                         | Disallow comparing `undefined` using `typeof`.                                                                                                                                                                    | ✅  | 🔧 | 💡 |
| [no-unnecessary-await](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/no-unnecessary-await.md)                                       | Disallow awaiting non-promise values.                                                                                                                                                                             | ✅  | 🔧 |    |
| [no-unreadable-array-destructuring](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/no-unreadable-array-destructuring.md)             | Disallow unreadable array destructuring.                                                                                                                                                                          | ✅  | 🔧 |    |
| [no-unreadable-iife](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/no-unreadable-iife.md)                                           | Disallow unreadable IIFEs.                                                                                                                                                                                        | ✅  |    |    |
| [no-unused-properties](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/no-unused-properties.md)                                       | Disallow unused object properties.                                                                                                                                                                                |    |    |    |
| [no-useless-fallback-in-spread](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/no-useless-fallback-in-spread.md)                     | Disallow useless fallback when spreading in object literals.                                                                                                                                                      | ✅  | 🔧 |    |
| [no-useless-length-check](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/no-useless-length-check.md)                                 | Disallow useless array length check.                                                                                                                                                                              | ✅  | 🔧 |    |
| [no-useless-promise-resolve-reject](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/no-useless-promise-resolve-reject.md)             | Disallow returning/yielding `Promise.resolve/reject()` in async functions or promise callbacks                                                                                                                    | ✅  | 🔧 |    |
| [no-useless-spread](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/no-useless-spread.md)                                             | Disallow unnecessary spread.                                                                                                                                                                                      | ✅  | 🔧 |    |
| [no-useless-switch-case](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/no-useless-switch-case.md)                                   | Disallow useless case in switch statements.                                                                                                                                                                       | ✅  |    | 💡 |
| [no-useless-undefined](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/no-useless-undefined.md)                                       | Disallow useless `undefined`.                                                                                                                                                                                     | ✅  | 🔧 |    |
| [no-zero-fractions](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/no-zero-fractions.md)                                             | Disallow number literals with zero fractions or dangling dots.                                                                                                                                                    | ✅  | 🔧 |    |
| [number-literal-case](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/number-literal-case.md)                                         | Enforce proper case for numeric literals.                                                                                                                                                                         | ✅  | 🔧 |    |
| [numeric-separators-style](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/numeric-separators-style.md)                               | Enforce the style of numeric separators by correctly grouping digits.                                                                                                                                             | ✅  | 🔧 |    |
| [prefer-add-event-listener](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/prefer-add-event-listener.md)                             | Prefer `.addEventListener()` and `.removeEventListener()` over `on`-functions.                                                                                                                                    | ✅  | 🔧 |    |
| [prefer-array-find](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/prefer-array-find.md)                                             | Prefer `.find(…)` and `.findLast(…)` over the first or last element from `.filter(…)`.                                                                                                                            | ✅  | 🔧 | 💡 |
| [prefer-array-flat](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/prefer-array-flat.md)                                             | Prefer `Array#flat()` over legacy techniques to flatten arrays.                                                                                                                                                   | ✅  | 🔧 |    |
| [prefer-array-flat-map](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/prefer-array-flat-map.md)                                     | Prefer `.flatMap(…)` over `.map(…).flat()`.                                                                                                                                                                       | ✅  | 🔧 |    |
| [prefer-array-index-of](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/prefer-array-index-of.md)                                     | Prefer `Array#{indexOf,lastIndexOf}()` over `Array#{findIndex,findLastIndex}()` when looking for the index of an item.                                                                                            | ✅  | 🔧 | 💡 |
| [prefer-array-some](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/prefer-array-some.md)                                             | Prefer `.some(…)` over `.filter(…).length` check and `.{find,findLast}(…)`.                                                                                                                                       | ✅  | 🔧 | 💡 |
| [prefer-at](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/prefer-at.md)                                                             | Prefer `.at()` method for index access and `String#charAt()`.                                                                                                                                                     | ✅  | 🔧 | 💡 |
| [prefer-blob-reading-methods](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/prefer-blob-reading-methods.md)                         | Prefer `Blob#arrayBuffer()` over `FileReader#readAsArrayBuffer(…)` and `Blob#text()` over `FileReader#readAsText(…)`.                                                                                             | ✅  |    |    |
| [prefer-code-point](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/prefer-code-point.md)                                             | Prefer `String#codePointAt(…)` over `String#charCodeAt(…)` and `String.fromCodePoint(…)` over `String.fromCharCode(…)`.                                                                                           | ✅  |    | 💡 |
| [prefer-date-now](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/prefer-date-now.md)                                                 | Prefer `Date.now()` to get the number of milliseconds since the Unix Epoch.                                                                                                                                       | ✅  | 🔧 |    |
| [prefer-default-parameters](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/prefer-default-parameters.md)                             | Prefer default parameters over reassignment.                                                                                                                                                                      | ✅  | 🔧 | 💡 |
| [prefer-dom-node-append](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/prefer-dom-node-append.md)                                   | Prefer `Node#append()` over `Node#appendChild()`.                                                                                                                                                                 | ✅  | 🔧 |    |
| [prefer-dom-node-dataset](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/prefer-dom-node-dataset.md)                                 | Prefer using `.dataset` on DOM elements over calling attribute methods.                                                                                                                                           | ✅  | 🔧 |    |
| [prefer-dom-node-remove](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/prefer-dom-node-remove.md)                                   | Prefer `childNode.remove()` over `parentNode.removeChild(childNode)`.                                                                                                                                             | ✅  | 🔧 | 💡 |
| [prefer-dom-node-text-content](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/prefer-dom-node-text-content.md)                       | Prefer `.textContent` over `.innerText`.                                                                                                                                                                          | ✅  |    | 💡 |
| [prefer-event-target](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/prefer-event-target.md)                                         | Prefer `EventTarget` over `EventEmitter`.                                                                                                                                                                         | ✅  |    |    |
| [prefer-export-from](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/prefer-export-from.md)                                           | Prefer `export…from` when re-exporting.                                                                                                                                                                           | ✅  | 🔧 | 💡 |
| [prefer-includes](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/prefer-includes.md)                                                 | Prefer `.includes()` over `.indexOf()` and `Array#some()` when checking for existence or non-existence.                                                                                                           | ✅  | 🔧 | 💡 |
| [prefer-json-parse-buffer](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/prefer-json-parse-buffer.md)                               | Prefer reading a JSON file as a buffer.                                                                                                                                                                           |    | 🔧 |    |
| [prefer-keyboard-event-key](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/prefer-keyboard-event-key.md)                             | Prefer `KeyboardEvent#key` over `KeyboardEvent#keyCode`.                                                                                                                                                          | ✅  | 🔧 |    |
| [prefer-logical-operator-over-ternary](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/prefer-logical-operator-over-ternary.md)       | Prefer using a logical operator over a ternary.                                                                                                                                                                   | ✅  |    | 💡 |
| [prefer-math-trunc](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/prefer-math-trunc.md)                                             | Enforce the use of `Math.trunc` instead of bitwise operators.                                                                                                                                                     | ✅  | 🔧 | 💡 |
| [prefer-modern-dom-apis](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/prefer-modern-dom-apis.md)                                   | Prefer `.before()` over `.insertBefore()`, `.replaceWith()` over `.replaceChild()`, prefer one of `.before()`, `.after()`, `.append()` or `.prepend()` over `insertAdjacentText()` and `insertAdjacentElement()`. | ✅  | 🔧 |    |
| [prefer-modern-math-apis](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/prefer-modern-math-apis.md)                                 | Prefer modern `Math` APIs over legacy patterns.                                                                                                                                                                   | ✅  | 🔧 |    |
| [prefer-module](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/prefer-module.md)                                                     | Prefer JavaScript modules (ESM) over CommonJS.                                                                                                                                                                    | ✅  | 🔧 | 💡 |
| [prefer-native-coercion-functions](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/prefer-native-coercion-functions.md)               | Prefer using `String`, `Number`, `BigInt`, `Boolean`, and `Symbol` directly.                                                                                                                                      | ✅  | 🔧 |    |
| [prefer-negative-index](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/prefer-negative-index.md)                                     | Prefer negative index over `.length - index` when possible.                                                                                                                                                       | ✅  | 🔧 |    |
| [prefer-node-protocol](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/prefer-node-protocol.md)                                       | Prefer using the `node:` protocol when importing Node.js builtin modules.                                                                                                                                         | ✅  | 🔧 |    |
| [prefer-number-properties](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/prefer-number-properties.md)                               | Prefer `Number` static properties over global ones.                                                                                                                                                               | ✅  | 🔧 | 💡 |
| [prefer-object-from-entries](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/prefer-object-from-entries.md)                           | Prefer using `Object.fromEntries(…)` to transform a list of key-value pairs into an object.                                                                                                                       | ✅  | 🔧 |    |
| [prefer-optional-catch-binding](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/prefer-optional-catch-binding.md)                     | Prefer omitting the `catch` binding parameter.                                                                                                                                                                    | ✅  | 🔧 |    |
| [prefer-prototype-methods](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/prefer-prototype-methods.md)                               | Prefer borrowing methods from the prototype instead of the instance.                                                                                                                                              | ✅  | 🔧 |    |
| [prefer-query-selector](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/prefer-query-selector.md)                                     | Prefer `.querySelector()` over `.getElementById()`, `.querySelectorAll()` over `.getElementsByClassName()` and `.getElementsByTagName()`.                                                                         | ✅  | 🔧 |    |
| [prefer-reflect-apply](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/prefer-reflect-apply.md)                                       | Prefer `Reflect.apply()` over `Function#apply()`.                                                                                                                                                                 | ✅  | 🔧 |    |
| [prefer-regexp-test](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/prefer-regexp-test.md)                                           | Prefer `RegExp#test()` over `String#match()` and `RegExp#exec()`.                                                                                                                                                 | ✅  | 🔧 | 💡 |
| [prefer-set-has](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/prefer-set-has.md)                                                   | Prefer `Set#has()` over `Array#includes()` when checking for existence or non-existence.                                                                                                                          | ✅  | 🔧 | 💡 |
| [prefer-set-size](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/prefer-set-size.md)                                                 | Prefer using `Set#size` instead of `Array#length`.                                                                                                                                                                | ✅  | 🔧 |    |
| [prefer-spread](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/prefer-spread.md)                                                     | Prefer the spread operator over `Array.from(…)`, `Array#concat(…)`, `Array#{slice,toSpliced}()` and `String#split('')`.                                                                                           | ✅  | 🔧 | 💡 |
| [prefer-string-replace-all](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/prefer-string-replace-all.md)                             | Prefer `String#replaceAll()` over regex searches with the global flag.                                                                                                                                            | ✅  | 🔧 |    |
| [prefer-string-slice](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/prefer-string-slice.md)                                         | Prefer `String#slice()` over `String#substr()` and `String#substring()`.                                                                                                                                          | ✅  | 🔧 |    |
| [prefer-string-starts-ends-with](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/prefer-string-starts-ends-with.md)                   | Prefer `String#startsWith()` & `String#endsWith()` over `RegExp#test()`.                                                                                                                                          | ✅  | 🔧 | 💡 |
| [prefer-string-trim-start-end](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/prefer-string-trim-start-end.md)                       | Prefer `String#trimStart()` / `String#trimEnd()` over `String#trimLeft()` / `String#trimRight()`.                                                                                                                 | ✅  | 🔧 |    |
| [prefer-switch](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/prefer-switch.md)                                                     | Prefer `switch` over multiple `else-if`.                                                                                                                                                                          | ✅  | 🔧 |    |
| [prefer-ternary](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/prefer-ternary.md)                                                   | Prefer ternary expressions over simple `if-else` statements.                                                                                                                                                      | ✅  | 🔧 |    |
| [prefer-top-level-await](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/prefer-top-level-await.md)                                   | Prefer top-level await over top-level promises and async function calls.                                                                                                                                          | ✅  |    | 💡 |
| [prefer-type-error](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/prefer-type-error.md)                                             | Enforce throwing `TypeError` in type checking conditions.                                                                                                                                                         | ✅  | 🔧 |    |
| [prevent-abbreviations](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/prevent-abbreviations.md)                                     | Prevent abbreviations.                                                                                                                                                                                            | ✅  | 🔧 |    |
| [relative-url-style](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/relative-url-style.md)                                           | Enforce consistent relative URL style.                                                                                                                                                                            | ✅  | 🔧 | 💡 |
| [require-array-join-separator](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/require-array-join-separator.md)                       | Enforce using the separator argument with `Array#join()`.                                                                                                                                                         | ✅  | 🔧 |    |
| [require-number-to-fixed-digits-argument](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/require-number-to-fixed-digits-argument.md) | Enforce using the digits argument with `Number#toFixed()`.                                                                                                                                                        | ✅  | 🔧 |    |
| [require-post-message-target-origin](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/require-post-message-target-origin.md)           | Enforce using the `targetOrigin` argument with `window.postMessage()`.                                                                                                                                            |    |    | 💡 |
| [string-content](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/string-content.md)                                                   | Enforce better string content.                                                                                                                                                                                    |    | 🔧 | 💡 |
| [switch-case-braces](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/switch-case-braces.md)                                           | Enforce consistent brace style for `case` clauses.                                                                                                                                                                | ✅  | 🔧 |    |
| [template-indent](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/template-indent.md)                                                 | Fix whitespace-insensitive template indentation.                                                                                                                                                                  | ✅  | 🔧 |    |
| [text-encoding-identifier-case](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/text-encoding-identifier-case.md)                     | Enforce consistent case for text encoding identifiers.                                                                                                                                                            | ✅  | 🔧 | 💡 |
| [throw-new-error](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/throw-new-error.md)                                                 | Require `new` when throwing an error.                                                                                                                                                                             | ✅  | 🔧 |    |

**Comment by ematipico** [comment]
- we already have a similar discussion if you could follow up that one instead of opening a new one https://github.com/rome/tools/discussions/3357
- we already have implemented some of those rules
- it's best to justify the reason why you want to propose (and maybe implement) new rules, so we have some context and reasoning

**Reply by Conaclos** [reply]
I completely agree with your point.
Many rules of unicorn are too pedantic or bring questionable benefits.
I think this go against the Rome philosophy.
We should have carefully chosen and consistent set of rules.

However, there are some valuable rules in the unicorn rule set.
Some of these rules are already implemented.

**Comment by flexiworld** [comment]
Frankly, I find that most of these rules add value to the linting process.
Sad to hear you find them too pedantic.

**Reply by Conaclos** [reply]
If i got time and motivation I will detail here which rule in my opinion make sense and which one are too pedantic in my taste.

**Reply by Conaclos** [reply]
I take a quick look and I must admit that most of the rules make sense.
I added some of them in #3357 under the `Unicorn` section.

I am a bit concerned by the number of rules to implement. This creates a cognitive load for the users. I think we should try to merge some of them. I am thinking about the rules that encourage sensible uses of native Array and String methods. These rules also enable discoverability.

Some rules such as `escape-case` could be part of the formatter.

I am also a bit concerned about DOM and node related rules. I think we should delay such rules and wait for a proper way to declare in which environment (node, worker, dom) a module is.

I think the following rules are too opinionated:
- [expiring-todo-comments](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/expiring-todo-comments.md)
- [no-null](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/no-null.md)
- [no-typeof-undefined](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/no-typeof-undefined.md)
- [no-nested-ternary](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/no-nested-ternary.md)
- [no-zero-fractions](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/no-zero-fractions.md)
- [prefer-math-trunc](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/prefer-math-trunc.md)
- [prefer-ternary](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/prefer-ternary.md)
- [prevent-abbreviations](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/prevent-abbreviations.md)
- [throw-new-error](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/throw-new-error.md)
 What about re-throwing?
- In progress...

I think the following rules are too narrow in scope:
- [no-object-as-default-parameter](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/no-object-as-default-parameter.md)
- [prefer-set-has](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/prefer-set-has.md)
- [prefer-set-size](https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/docs/rules/prefer-set-size.md)
- In progress...
